packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts

```diff
@@ -1531,3 +1531,43 @@ test('combine mappings', async () => {
     `)
   }
 })
+
+test('deps', async () => {
```

Comment:
This is ported from oxc https://github.com/oxc-project/oxc/blob/932a66c0b4fb689d72d763bcade1c6e8340e4ca8/crates/oxc_transformer/src/plugins/module_runner_transform.rs#L2267